DataFormats/SiStripCluster/interface/SiStripCluster.h

```diff
@@ -58,13 +51,19 @@ class SiStripCluster {
    *  You can find the special meanings of values { 0, 254, 255} in section 3.4.1 of
    *  http://www.te.rl.ac.uk/esdg/cms-fed/firmware/Documents/FE_FPGA_Technical_Description.pdf
    */
-  uint8_t const* begin() const { return amplitudes_.data(); }
-  uint8_t const* end() const { return begin() + size(); }
-  uint8_t size() const { return amplitudes_.size(); }
-  uint8_t operator[](int i) const { return *(begin() + i); }
+  auto size() const { return amplitudes_.size(); }
+  auto const* begin() const { return amplitudes_.data(); }
+  auto const* end() const { return begin() + size(); }
+  auto operator[](int i) const { return *(begin() + i); }
```

Comment:
could I suggest to _not_ use `auto` for the return type of a struct/class public methods ?
it makes it a lot harder to figure out what the return type actually is :-(

Reply:
this is forwarding the type used by a std container that is a ```typedef``` inside the container itself.
better ```auto``` than a wrong guess.
the alternative is to use ```typename Container::guessed_typedef```

Reply:
But that means the _user_ is now left to guess the type... at least the developer of the class has to make the guess only once.

Reply:
why the "user" shall guess the type?  Just use ```auto``` or ```decltype```.
what users do with std container?

Reply:
what about keeping `auto`, but putting a comment about the return type? Anyway @cms-sw/reconstruction-l2 please state your preference here, in order to move on.

Reply:
I would prefer to go in the direction of `auto`, considering that the bug was coming from a mismatch between the real type and the one put by the dev.
`auto` will always catch `typename Container<T>::size_type`

Reply:
Sorry, I don't understand the argument, or maybe I did not understand the bug and the fix.

`amplitudes_` is a `std::vector<uint8_t>`, so `amplitudes_.data()` is an `uint8_t *`, and `auto const* begin() const` returns a `uint8_t const*`, which is the type that was explicitly used in the interface.

Same for `end()`.

Same for `operator[](int i)` - though it could be simplified to just return `amplitudes_[i]`.

So the only fix is in the return type of `size()` ?
Yes, returning an `uint8_t` is definitely wrong... why not change _that_ to `auto` (or `size_t`, or `vector<uint8_t>::size_type`) and keep the other types explicit, so people don't have to read the whole file to see what type is being returned ?